### DISCUSSION TITLE
Request path should be agnostic?

### DISCUSSION THREAD

**Discussion by nineunderground** [coder/code-server#5674] [General] [answered]
Hi,

Just trying to run several containers behind a reverse proxy like Nginx. Thus, I can add rules to forward to them base on path. The first request goes well, but then some followed requests that require static files, javascript files, etc.. are removing the base path, so those cannot reach the container.

As I understand base-path or base-url parameter was removed, so how this approach can be done?

Thanks in advance

**Comment by nineunderground** [comment]
Not sure now if this is even a bug like it was mentioned in the past?
https://github.com/coder/code-server/issues/241

**Comment by nineunderground** [comment, marked as answer]
I've found the issue.
So, it seems I need to add one more slash "/" at the end of every requests, otherwise code server removes the whole base URL.
Is that expected behavior?

I think from end user experience it doesn't make sense that...
_https://my-hostname:my-port/vscode1_ does not work, but
_https://my-hostname:my-port/vscode1/_ it does

**Reply by jsjoeio** [reply]
I believe it's expected that the user uses a trailing slash i they host on a subpath  🤔 maybe @code-asher remembers

**Reply by code-asher** [reply]
Yeah this is unfortunately just how it works because of how browsers resolve relative URLs.  

`/vscode/` + `./some/path` == `/vscode/some/path`
`/vscode` + `./some/path` == `/some/path`

When you use `/vscode1` or `/vscode1/` NGINX rewrites both of them to `/` so code-server has no way of knowing which it is.

Some applications work around this by asking you for the full base path but since code-server uses relative URLs now instead the trailing slash is required.

**Reply by code-asher** [reply]
We might want to add this to the docs or add an automatic redirect from no trailing slash to trailing slash to our example configs.
